pkg/core/xds/resource.go

```diff
@@ -1,25 +1,22 @@
 package xds
 
 import (
-	"github.com/golang/protobuf/proto"
-	"github.com/golang/protobuf/ptypes"
+	"sort"
 
 	envoy "github.com/envoyproxy/go-control-plane/envoy/api/v2"
 	envoy_types "github.com/envoyproxy/go-control-plane/pkg/cache/types"
+	envoy_resource "github.com/envoyproxy/go-control-plane/pkg/resource/v2"
+	"github.com/golang/protobuf/proto"
+	"github.com/golang/protobuf/ptypes"
 )
 
 // ResourcePayload is a convenience type alias.
 type ResourcePayload = envoy_types.Resource
 
-type NamedResourcePayload interface {
-	ResourcePayload
-	GetName() string
-}
-
 // Resource represents a generic xDS resource with name and version.
 type Resource struct {
 	Name     string
-	Version  string
+	Origin   string
```

Comment:
Why removing the `version`?

Reply:
Version is useless here. We don't use it anyways, we override version anyways (see `SnapshotAutoVersioner`)